mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -124,7 +130,26 @@ typedef struct _ScanData {
 	unsigned obj_state : 2;
 } ScanData;
 
+#define HEAVY_REFS_MIN 2
+#define HEAVY_COMBINED_REFS_MIN 60
```

Comment:
This is slightly arbitrary and I do not have any complex Monodroid programs handy to fine-tune values against. I picked it such that if the incoming_colors ever hits its maximum size, the HEAVY_COMBINED_REFS_MIN would be sure to trigger. I note that if you hit the HEAVY_COMBINED_REFS_MIN then that means you are generating half a kilobyte of xrefs just for the list.